strickland/src/compare.js

```diff
@@ -22,7 +22,7 @@ export default function compareValidator(validatorProps) {
 
         let isValid = true;
 
-        if (isFalsyButNotZero(value)) {
+        if (typeof compare !== 'boolean' && isFalsyButNotZero(value)) {
```

Comment:
Would you mind adding a couple unit tests for this behavior?  And if it's not too much to ask, updating the docs for `compare` to call this handling out as well.